### ISSUE TITLE
Handle additionals attributes in asvc schema declaration

### ISSUE THREAD

**Issue by M3lkior** [asyncapi/avro-schema-parser#18]
#### Reason/Context
Please try answering few of those questions
- Why we need this improvement?

The Avro specification is limited by a restricted attributes list in field declaration.
For example, the avro spec not handle `example`, `min` or `max` attributes but it is possible to set those attributes in avro specification without breaking the declaration.

I think that the avro-parser should be consider the additional attributes matching with the async api specification like `example`, `min` and `max` for number, etc

- How will this change help?

By considering those fields, the generated files (HTML for example) should handle those infomations and display them in the generated template (like example values on attributes by example. Today, all values are set to null when using asvc files)

- What is the motivation? 

I use avro specification and html-template to generated AsyncApi documentations in my organization and it could be useful to display those informations.

#### Description
Please try answering few of those questions
- What changes have to be introduced?

Handle additional attributes, drived by the AsyncAPI specification
- Will this be a breaking change?

Nop

- How could it be implemented/designed?

Check if the avro specification contains the following attributes, and include them in the JSON schema if true : 
 
- maximum
- exclusiveMaximum
- minimum
- exclusiveMinimum
- maxLength
- minLength
- pattern (This string SHOULD be a valid regular expression, according to the ECMA 262 regular expression dialect)
- maxItems
- minItems
- examples
-

**Comment by derberg** [comment]
@M3lkior is this one still valid?

**Comment by M3lkior** [comment]
Yes.

**Comment by derberg** [comment]
@M3lkior something you think you can pick up 😉

**Comment by dalelane** [comment]
>  it is possible to set those attributes in avro specification without breaking the declaration

@M3lkior Can you expand on this a bit, please? 

I'm curious exactly what you mean. Are you saying that the Avro specification explicitly tolerates additional attributes that aren't defined in the spec? Or that some specific tooling based on Avro tolerates unexpected attributes? Or something else?

**Comment by M3lkior** [comment]
> > it is possible to set those attributes in avro specification without breaking the declaration
> 
> @M3lkior Can you expand on this a bit, please?
> 
> I'm curious exactly what you mean. Are you saying that the Avro specification explicitly tolerates additional attributes that aren't defined in the spec? Or that some specific tooling based on Avro tolerates unexpected attributes? Or something else?

Yes this is exactly what i want to explain. The idea behind this capability is to permit displaying additional fields in the generated docs ; like the `example` or `min` / `max` fields. 

Given this kind of specification (which not break the AVRO maven plugin for example)

```
{
    "namespace": "com.adeo.ccr.costing.exchange.avro.model.costingrequest",
    "type": "record",
    "name": "CostingRequestPayload",
    "version": 1,
    "fields": [
        {"name": "Length", "type": ["int"], "example": "10", "min": "0", "max": "1"}
]}
```

Would allow the parser to display the `example` ; `min` and `max` attributes in the generated doc which is not the case today : 
![image](https://user-images.githubusercontent.com/5501911/117947620-a8eec100-b310-11eb-897e-d9aad28acee4.png)

**Comment by derberg** [comment]
I tried to find some answers in Avro spec documentation but failed. I could not find any information about how they approach additional attributed, could not find a JSON Schema for the spec too. Do you know how to contact Avro folks? do they use Slack or something?

**Comment by ITman1** [comment]
@derberg Is not that mentioned at the beginning of [Avro spec](https://avro.apache.org/docs/current/spec.html)?

> A JSON object, of the form: {"type": "typeName" ...attributes...} where typeName is either a primitive or derived type name, as defined below. **Attributes not defined in this document are permitted as metadata**, but must not affect the format of serialized data.

No contacts to Avro folks so far, but maybe @iemejia could confirm, he is quite regular recent committer. 😄

**Comment by derberg** [comment]
@ITman1 it does sound like that but I'm not native speaker and would prefer to see JSON Schema with `additionalProperties` set to true 😅 although I'm not a machine 😄

**Comment by derberg** [comment]
Ok, I finally found one here https://json.schemastore.org/avro-avsc.json
I hope it is an official one 😅 

I do not see `"additionalProperties": false` there which means to me they are allowed. Lemme go back to the PR

**Comment by M3lkior** [comment]
> Ok, I finally found one here https://json.schemastore.org/avro-avsc.json
> I hope it is an official one 😅
> 
> I do not see `"additionalProperties": false` there which means to me they are allowed. Lemme go back to the PR

@derberg i start working on this feature this week for information

**Comment by M3lkior** [comment]
i work on them as hacktoberfest issue :)
